bravado/requests_client.py

```diff
@@ -185,18 +218,55 @@ def check_for_exceptions(self, response):
         except Exception as e:
             add_response_detail_to_errors(e)
 
+    def build_timeout(self, result_timeout):
+        """
+        Build the appropriate timeout object to pass to `session.send(...)`.
+
+        :param result_timeout: timeout that was passed into `result(..)`.
+        :return: timeout
+        :rtype: float or tuple(connect_timeout, timeout)
+        """
+        timeout = None
+        has_service_timeout = 'timeout' in self.misc_options
+        has_result_timeout = result_timeout is not None
```

Comment:
Well, that's fair point. But from `timeout`'s [docstring](http://bravado.readthedocs.org/en/latest/bravado.html#bravado.http_future.HttpFuture.result) behavior with `None` is undefined. I think with a default value provided, a user given `None` value is really a special case than some accepted scenario.